### ISSUE TITLE
GDShader "invalid pragma directive" error on unrecognized pragma should be dropped

### ISSUE THREAD

**Issue by geekley** [godotengine/godot#97738]
### Tested versions

- Reproducible in: v4.3.stable.flathub [77dcf97d8]

### System information

Godot v4.3.stable (77dcf97d8) - Freedesktop SDK 23.08 (Flatpak runtime) - X11 - Vulkan (Forward+) - integrated Intel(R) HD Graphics 5500 (BDW GT2) - Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz (4 Threads)

### Issue description

The [GLSL ES 3.00 spec](https://registry.khronos.org/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf) says on page 15:
> `#pragma` allows implementation dependent compiler control. Tokens following `#pragma` are not subject to preprocessor macro expansion. **If an implementation does not recognize the tokens following `#pragma`, then it will ignore that pragma.**

I believe the whole point of `#pragma` is to allow forward-compatibility, by letting any unrecognized instructions be ignored. So I believe showing an error like "Invalid pragma directive" defeats the whole purpose, no? Invalid pragma lines should be ignored like how it is in the spec.

In fact, both C and GLSL preprocessors don't complain at all if you add a custom pragma.

### Steps to reproduce

```glsl
shader_type spatial;
#pragma some_custom "etc" 3.14 // unrecognized pragma should be ignored instead of raising error
```
```sh
cd /tmp
echo '#pragma some_custom "etc" 3.14' | tee a.c > a.frag
echo '=== GLSL ==='; glslang -E a.frag; echo '=== C ==='; gcc -E a.c
```

Of course, the error should still show for recognized pragma lines that are malformed:
```glsl
#pragma disable_preprocessor blah blah // should still raise the error
```

### Minimal reproduction project (MRP)

N/A

**Comment by dustdfg** [comment]
It is mostly just a game of words but godot doesn't use GLSL ES 3.0. It uses it's own language: [Godot shading language](https://docs.godotengine.org/en/stable/tutorials/shaders/shader_reference/shading_language.html#doc-shading-language)... Which is yeah very similar to glsl

Anyway effectively godot doesn't have any "useful" pragmas and uses it [only for disabling preprocessor](https://docs.godotengine.org/en/stable/tutorials/shaders/shader_reference/shader_preprocessor.html#pragma) so I agree that throwing error isn't a good approach.

**Comment by geekley** [comment]
@dustdfg Yes, but GDShader is based on GLSL ES 3.0, and it's been decided that the preprocessor behavior should aim to be GLSL-compatible where possible: https://github.com/godotengine/godot/issues/96253#issuecomment-2333734897
> I brought up this Issue in the 2024-09-03 Rendering Meeting, these are the key points:
> - Aim for glsl-like behavior, so that shaders can be ported over very easily [...]

**Comment by Chaosus** [comment]
This issue should probably be closed, since @clayjohn don't like it.

**Comment by geekley** [comment]
I left additional comments in the PR to further explain my arguments in favor of this: https://github.com/godotengine/godot/pull/98413#issuecomment-2551478489, https://github.com/godotengine/godot/pull/98413#issuecomment-2551550473

If you still disagree, may I ask that you please turn it into a discussion instead of closing it?
